src/pages/ReferralDetailsPage.js

```diff
@@ -0,0 +1,107 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import Button from '@components/Button';
+import CopyTextToClipboard from '@components/CopyTextToClipboard';
+import FixedFooter from '@components/FixedFooter';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import Icon from '@components/Icon';
+import {PaymentHands} from '@components/Icon/Illustrations';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import TextLink from '@components/TextLink';
+import useLocalize from '@hooks/useLocalize';
+import Navigation from '@libs/Navigation/Navigation';
+import styles from '@styles/styles';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import NotFoundPage from './ErrorPage/NotFoundPage';
+
+const propTypes = {
+    /** Navigation route context info provided by react navigation */
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            /** The type of the content from where CTA was called */
+            contentType: PropTypes.string,
+        }),
+    }).isRequired,
+
+    /** The details about the account that the user is signing in with */
+    account: PropTypes.shape({
+        /** Whether or not a sign on form is loading (being submitted) */
+        loading: PropTypes.bool,
+
+        /** Whether or not the account is validated */
+        validated: PropTypes.bool,
+
+        /** The primaryLogin associated with the account */
+        primaryLogin: PropTypes.string,
+    }),
+};
+
+const defaultProps = {
+    account: null,
+};
+
+function ReferralDetailsPage({route, account}) {
+    const {translate} = useLocalize();
+    const {contentType} = route.params;
+    if (!_.includes(_.values(CONST.REFERRAL_PROGRAM.CONTENT_TYPES), contentType)) {
+        return <NotFoundPage />;
+    }
+    const contentHeader = translate(`referralProgram.${contentType}.header`);
+    const contentBody = translate(`referralProgram.${contentType}.body`);
+    const generatedURL = `${CONST.REFERRAL_PROGRAM.LINK}/?thanks=${account && account.primaryLogin}`;
```

Comment:
This lacks character escaping!

Reply:
Also, could be extracted to a separate function, at least

Reply:
you would like to have a separate function for generatedURL?

What do you mean by this: "This lacks character escaping!"?

Reply:
> you would like to have a separate function for generatedURL?

I think this would be slightly more readable.

> What do you mean by this: "This lacks character escaping!"?

Well, this wasn't phrased accurately. We're not talking about character escaping, but rather URL encoding.

You can't just concatenate a URL prefix with an arbitrary string, as some characters in that arbitrary string may be _special_ characters in the URL format. `+` is just an example. It's a valid character in e-mail, while in an URL it represents a space.

Character escaping is a solution to a similar problem in some languages/data formats, for example in `"foo\"bar"`, the inner `"` is escaped.

Reply:
Okay, get it.
Something like this should work:
```javascript
function generateURL(email) {
    const baseURL = `${CONST.REFERRAL_PROGRAM.LINK}/?thanks=`;
    const encodedEmail = encodeURIComponent(email);
    const generatedURL = baseURL + encodedEmail;
    return generatedURL;
}
```

Reply:
Yeah, that will do. The second option is fine. 👍

While I know it's only a single expression, it's quite different functionally from the surrounding code; that's why I'd extract it.

Reply:
Updated @cubuspl42